### PR TITLE
Fix-80080 Show more detailed error message for "Regex parse error" in search

### LINKED ISSUE #80080: Show more detailed error message for "Regex parse error" in search
See https://github.com/microsoft/vscode/issues/79109#issuecomment-521315899

We can parse the full error details from the ripgrep error message to show more details for the error message.

### PR DESCRIPTION
**Pull request by skprabhanjan**
@roblourens , Here is the PR to close #80080.
I thought the below format would be good.

**"Regex Parse Error"
"regex could not be compiled with either the default regex engine or with PCRE2."
"error: " related to default regex
"PCRE2: " related to PCRE2 .**

I need help on one more thing which I am not able to figure out.
**By adding "\n" , the contents are not coming up in new line !**

### CODE REVIEW

**Comment by skprabhanjan** [comment]
Screen Shot 2019-09-06 at 9 45 43 PM

**Review by roblourens — CHANGES_REQUESTED** [review]
Thanks for the PR, this is a good start. What I would like to see:

- Only show the PCRE2 error message, without any "PCRE2" label
- No newlines
- Test cases

**Comment by skprabhanjan** [comment]
@roblourens , All changes done according to the review :)
PS : I have added **( ; ) as the separator** for the strings ( Looked good and more clean ) 
I can change it something else if you suggest :)

**Review by roblourens — COMMENTED** [review]
Thanks, but I don't want to see the error message from the non-PCRE2 engine at all. These are just confusing implementation details that the user doesn't care about. It shouldn't show anything about "engines" or "compiling" or anything else besides the error that the PCRE2 engine gave. To me, this is the "real" error message.

**Comment by skprabhanjan** [comment]
@roblourens , The changes are done .
Please let me know if there are any concerns :)

File: src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts [hunk 1]
```diff
@@ -150,6 +150,20 @@ export function rgErrorMsgForDisplay(msg: string): Maybe<SearchError> {
 	return undefined;
 }
 
+export function buildRegexParseError(lines: string[]): string {
+	let errorMessage: string[] = ['Regex parse error'];
+	lines.forEach(line => {
```
Comment (roblourens) [inline src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts]: This loop makes it look like you are handling multiple lines that start with PCRE2:. Does that ever happen? I don't see a case where that happens. Unless you know of a case where that happens I would prefer it to just handle the first one it finds. 

The error message should just be formatted like `Regex parse error: error message` (no extra spaces or colons)

Comment (skprabhanjan) [reply src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts]: Done

**Comment by skprabhanjan** [comment]
@roblourens , Sorry for these little misses.
Its all done hopefully now :) 
Please review this :)

File: src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts [hunk 2]
```diff
@@ -152,15 +152,11 @@ export function rgErrorMsgForDisplay(msg: string): Maybe<SearchError> {
...
-	lines.forEach(line => {
-		if (startsWith(line, 'PCRE2:')) {
-			// Just show the error without PCRE2 label
-			let pcre2ErrorMessage = line.replace('PCRE2:', '');
-			let pcre2ActualErrorMessage = pcre2ErrorMessage.split(':')[1];
-			errorMessage.push(pcre2ActualErrorMessage);
-		}
-	});
-	return errorMessage.join(' : ');
+	let pcre2ErrorLine = lines.filter(l => (startsWith(l, 'PCRE2:')))[0];
```
Comment (roblourens) [inline src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts]: This is better but `pcre2ErrorLine` will be undefined if there is no matching line. It shouldn't fail if the string is in an unexpected format.

File: src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts [hunk 3]
```diff
@@ -152,15 +152,11 @@ export function rgErrorMsgForDisplay(msg: string): Maybe<SearchError> {
...
+	let pcre2ErrorMessage = pcre2ErrorLine.replace('PCRE2:', '');
+	let pcre2ActualErrorMessage = pcre2ErrorMessage.split(':')[1];
```
Comment (roblourens) [inline src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts]: And this shouldn't fail if there is no `:`. If we don't parse details out of the format, then it should fall back on `Regex parse error`

**Review by roblourens — CHANGES_REQUESTED** [review]

**Comment by skprabhanjan** [comment]
@roblourens , I wanted to do that check but was not sure as I assumed the error will be in that format always but It is fixed now.

**Review by roblourens — APPROVED** [review]
Thanks!

**Pull request merged**